### PR TITLE
Simplify MessageList, fix bug with loading indicator

### PR DESCRIPTION
**Pull request by borisyankov**
* Merge message-list and message, current separation didn't make enough sense
* Extract 'message loading' into a separate component
* Improve readability and simplify how 'loading' and 'no messages' are determined
* Fix a bug with showing message loading incorrectly.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@borisyankov, it looks like you've already signed the Dropbox CLA. Thanks!

File: src/chat/Chat.js [hunk 1]
```diff
@@ -1,40 +1,25 @@
...
 import NoMessages from '../message/NoMessages';
 import ComposeBox from '../compose/ComposeBox';
 
 export default class MainScreen extends React.Component {
-
   render() {
-    const { auth, messages, narrow, fetching, caughtUp, subscriptions,
-      isOnline, twentyFourHourTime, doNarrow, fetchOlder, fetchNewer,
-      markAsRead, mute } = this.props;
+    const { messages, narrow, caughtUp, fetching, isOnline } = this.props;
```
Comment (neerajwahi) [inline src/chat/Chat.js]: Can we get rid of `Chat.js` too and just use `MessageList` from `MainScreen`? I feel like it adds an unnecessary level of indirection

File: src/chat/chatReducers.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ import { homeNarrow, isMessageInNarrow } from '../utils/narrow';
 import chatUpdater from './chatUpdater';
 
 const getInitialState = () => ({
-  fetching: { older: false, newer: false },
+  fetching: { older: true, newer: true },
```
Comment (neerajwahi) [inline src/chat/chatReducers.js]: Why change these to `true`?

File: src/message/MessageList.js [hunk 3]
```diff
@@ -101,15 +66,17 @@ export default class MessageList extends React.PureComponent {
...
         style={styles.list}
-        contentContainerStyle={containerStyle}
         automaticallyAdjustContentInset="false"
         stickyHeaderIndices={headerIndices}
         onStartReached={fetchOlder}
         onEndReached={fetchNewer}
         autoScrollToBottom={this.state.autoScrollToBottom}
         onScroll={this.onScroll}
       >
+        {hideFetchingOlder &&
```
Comment (neerajwahi) [inline src/message/MessageList.js]: Nice improvement!

File: src/chat/Chat.js (hunk 1 above)
Comment (borisyankov) [inline src/chat/Chat.js]: Maybe. I also had the sense we have one more wrappers than needed.

File: src/chat/chatReducers.js (hunk 2 above)
Comment (borisyankov) [inline src/chat/chatReducers.js]: Since we always do an initial fetch, and the code for initial fetch does not do an action call.

**Pull request merged**